### PR TITLE
[Collection], [Configuration], [Exception], [Executor] upgrade code to php7

### PR DESCRIPTION
**Pull request by rvitaliy**
| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | No
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | N/A

Improve code with php7 syntax

### CODE REVIEW

File: src/Configuration/Configuration.php [hunk 1]
```diff
@@ -27,7 +27,7 @@ public function __construct()
     /**
      * @return Collection
```
[the remark this answers is not in the document]
Comment (rvitaliy) [reply src/Configuration/Configuration.php]: i fully agree.
i do it ASAP.

**Comment by kick-the-bucket** [comment]
Looks like this PR broke custom `InputOption`s, which use the mode `InputOption::VALUE_NONE`.
I am getting `[TypeError] strlen() expects parameter 1 to be string, boolean given` on ParallelExecutor.php:265

**Comment by antonmedv** [comment]
@rvitaliy can you check this?

**Comment by rvitaliy** [comment]
@antonmedv @kick-the-bucket of course
I took a quick look at the code
it's not a bug it's a defect that emerged after we add `declare(strict_types=1);`
`InputInterface::getOptions` has multiple type of behavior `VALUE_NONE`, `VALUE_REQUIRED`, `VALUE_OPTIONAL`, VALUE_IS_ARRAY`
`ParallelExecutor::generateOptions` must manage all this types.

@kick-the-bucket if you can prepare a PR to manage all values, you are welcome. ASAP I find a couple of free hours I will be happy to help you solve the problem

**Comment by kick-the-bucket** [comment]
It probably never handled array options well, but it did use to work with VALUE_NONE options correctly before this change.

**Comment by antonmedv** [comment]
so can @kick-the-bucket create PR? And @rvitaliy review it? :)

**Comment by antonmedv** [comment]
Merged!

**Pull request merged**